Source/JavaScriptCore/bytecode/Repatch.cpp

```diff
@@ -665,6 +665,38 @@ static InlineCacheAction tryCacheArrayGetByVal(JSGlobalObject* globalObject, Cod
             case ArrayStorageShape:
                 accessType = AccessCase::IndexedArrayStorageLoad;
                 break;
+            case NoIndexingShape: {
+                if (!base->isObject())
+                    return GiveUpOnCache;
+
+                if (base->structure()->mayInterceptIndexedAccesses() || base->structure()->typeInfo().interceptsGetOwnPropertySlotByIndexEvenWhenLengthIsNotZero())
+                    return GiveUpOnCache;
+
+                auto cacheStatus = prepareChainForCaching(globalObject, base, nullptr);
+                if (!cacheStatus)
+                    return GiveUpOnCache;
+
+                if (cacheStatus->usesPolyProto)
+                    return GiveUpOnCache;
+
+                Structure* headStructure = base->structure();
+                ObjectPropertyConditionSet conditionSet = generateConditionsForIndexedMiss(vm, codeBlock, globalObject, headStructure);
+                if (!conditionSet.isValid())
+                    return GiveUpOnCache;
+
+                auto newCase = AccessCase::create(vm, codeBlock, AccessCase::IndexedNoIndexingMiss, nullptr, invalidOffset, headStructure, conditionSet);
+                result = stubInfo.addAccessCase(locker, globalObject, codeBlock, ECMAMode::strict(), nullptr, WTFMove(newCase));
+
+                if (result.generatedSomeCode()) {
+                    LOG_IC((ICEvent::GetByReplaceWithJump, baseValue.classInfoOrNull(), Identifier()));
+
+                    RELEASE_ASSERT(result.code());
+                    InlineAccess::rewireStubAsJumpInAccessNotUsingInlineAccess(codeBlock, stubInfo, CodeLocationLabel<JITStubRoutinePtrTag>(result.code()));
+                }
+
+                fireWatchpointsAndClearStubIfNeeded(vm, stubInfo, codeBlock, result);
+                return result.shouldGiveUpNow() ? GiveUpOnCache : RetryCacheLater;
```

Comment:
this is a lot of duplicated code for no reason. I suggest adopting the idiom we do in other repatch code where we have a unique_ptr on AccessCase, and if it's null, we call create below. If it's non-null, we don't call create. That way we don't have this copy pasted code here.

Reply:
That sounds good. will change.